api/src/main/java/com/epam/pipeline/controller/metadata/CategoricalAttributeController.java

```diff
@@ -0,0 +1,126 @@
+
+    @PostMapping
+    @ResponseBody
+    @ApiOperation(
+        value = "Add categorical attributes values.",
+        notes = "Add categorical attributes values",
+        produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiResponses(
+        value = {@ApiResponse(code = HTTP_STATUS_OK, message = API_STATUS_DESCRIPTION)
+        })
+    public Result<Boolean> insertCategoricalAttributes(@RequestBody final Map<String, List<String>> dict) {
+        return Result.success(categoricalAttributeApiService.insertAttributesValues(dict));
+    }
+
+    @GetMapping("/loadAll")
+    @ResponseBody
+    @ApiOperation(
+        value = "Load all categorical attributes with values.",
+        notes = "Load all categorical attributes with values.",
+        produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiResponses(
+        value = {@ApiResponse(code = HTTP_STATUS_OK, message = API_STATUS_DESCRIPTION)
+        })
+    public Result<Map<String, List<String>>> loadAllCategoricalAttributes() {
+        return Result.success(categoricalAttributeApiService.loadAll());
+    }
+
+    @GetMapping("/load")
+    @ResponseBody
+    @ApiOperation(
+        value = "Load all requested attributes with values.",
+        notes = "Load all requested attributes with values.",
+        produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiResponses(
+        value = {@ApiResponse(code = HTTP_STATUS_OK, message = API_STATUS_DESCRIPTION)
+        })
+    public Result<Map<String, List<String>>> loadCategoricalAttributes(@RequestParam final List<String> attributeKeys) {
+        return Result.success(categoricalAttributeApiService.loadAllValuesForKeys(attributeKeys));
+    }
+
+    @DeleteMapping("/delete")
+    @ResponseBody
+    @ApiOperation(
+        value = "Delete all values for a requested attribute.",
+        notes = "Delete all values for a requested attribute.",
+        produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiResponses(
+        value = {@ApiResponse(code = HTTP_STATUS_OK, message = API_STATUS_DESCRIPTION)
+        })
+    public Result<Boolean> deleteAllAttributeValues(@RequestParam final String attributeKey) {
+        return Result.success(categoricalAttributeApiService.deleteAttributeValues(attributeKey));
+    }
+
+    @DeleteMapping("/deleteValue")
+    @ResponseBody
+    @ApiOperation(
+        value = "Delete one specific value for a requested attribute.",
+        notes = "Delete one specific value for a requested attribute.",
+        produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiResponses(
+        value = {@ApiResponse(code = HTTP_STATUS_OK, message = API_STATUS_DESCRIPTION)
+        })
+    public Result<Boolean> deleteAttributeValue(@RequestParam final String attributeKey,
+                                                @RequestParam final String value) {
+        return Result.success(categoricalAttributeApiService.deleteAttributeValue(attributeKey, value));
+    }
+
+    @GetMapping("/sync")
```

Comment:
Let's make it POST, since this request is not idempotent

Reply:
Done